docs/src/app/components/AppBar/ExampleWaterfallOptimized.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import {AppBar} from 'material-ui';
+
+import IconButton from 'icon-button';
+import MoreVertIcon from 'svg-icons/navigation/more-vert';
+import ArrowBack from 'svg-icons/navigation/arrow-back';
+
+const MIN_HEIGHT = 64;
+const MAX_HEIGHT = 210;
+
+const AppBarWaterfallExample = React.createClass({
+
+  propTypes: {
+    onBack: React.PropTypes.func,
+  },
+
+  render() {
+    const styles = this.getStyles();
+    return (
+      <AppBar
+        position="waterfall"
+        waterfall={{
+          minHeight: MIN_HEIGHT,
+          maxHeight: MAX_HEIGHT,
+          onHeightChange: this.onHeightChange,
+          children: (<div
+            style={styles.logoWrap}>
+            <img
+              ref={el => { this.logoEl = el; }}
```

Comment:
Having to rely on a ref to optimize the component seems weird.
Can't we change the API. I think that it would be better to have a `children` property
as a function that take the height as an input and return a component.
This way, we would only need one example, a simple and optimized one.